packages/flutter/test/widgets/navigator_test.dart

```diff
@@ -628,6 +628,61 @@ void main() {
     expect(aOffset.dx, lessThan(aOffsetOriginal.dx));
   });
 
+  testWidgets('pushReplacement correctly reports didReplace to the observer', (WidgetTester tester) async {
+    // Regression https://github.com/flutter/flutter/issues/56892.
+    final Map<String, WidgetBuilder> routes = <String, WidgetBuilder>{
+      '/' : (BuildContext context) => const OnTapPage(
+        id: '/',
+      ),
+      '/A': (BuildContext context) => const OnTapPage(
+        id: 'A',
+      ),
+      '/A/B': (BuildContext context) => OnTapPage(
+        id: 'B',
+        onTap: (){
+          Navigator.of(context).popUntil((r) => r.isFirst);
+          Navigator.of(context).pushReplacementNamed('/C');
+        },
+      ),
+      '/C': (BuildContext context) => const OnTapPage(id: 'C',
+      ),
+    };
+    final List<List<Route<dynamic>>> didPopped = <List<Route<dynamic>>>[];
```

Comment:
Instead of having a list of list maybe define a helper class to make the test more readable: 

```dart
class RoutePair {
 const RoutePair({this.previous, this.current});
 final String previous;
 final String current;
}
```